sklego/meta/grouped_transformer.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from sklego.common import as_list
+
+
+class GroupedTransformer(BaseEstimator, TransformerMixin):
+    """
+    Construct a transformer per data group. Splits data by groups from single or multiple columns
+    and transforms remaining columns using the transformers corresponding to the groups.
+
+    :param transformer: the transformer to be applied per group
+    :param groups: the column(s) of the matrix/dataframe to select as a grouping parameter set
+    :param use_global_model: Whether or not to fall back to a general transformation in case a group
+                             is not found during `.transform()`
+    """
+
+    def __init__(self, transformer, groups=0, use_global_model=True):
```

Comment:
Groups is something that the user *must* supply. It'd be nicer if it weren't initialized with 0 with this in mind.

Reply:
I think `sklearn` expects that an estimator can be initialised without any arguments specified by the user, although I'm not sure how strictly that is enforced. would a `groups=None` make sense to have a no grouping at all option as a default?

Reply:
Ah yeah you're right. `None` would be better. That said, `meta` estimators *must* receive an estimator in order for it to work.